sdk/dapp-kit/src/components/styling/InjectedThemeStyle.tsx

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { assignInlineVars } from '@vanilla-extract/dynamic';
+
+import { styleDataAttributeSelector } from '../../constants/styleDataAttribute.js';
+import { themeVars } from '../../themes/themeContract.js';
+import type { Theme, ThemeVars } from '../../themes/themeContract.js';
+
+type InjectedThemeStyleProps = {
+	theme: Theme;
+};
+
+export function InjectedThemeStyle({ theme }: InjectedThemeStyleProps) {
+	const defaultStyles = `${styleDataAttributeSelector}{${cssStringFromTheme(
+		'light' in theme ? theme.light : theme,
+	)}}`;
+	const darkModeStyles =
+		'dark' in theme
+			? `@media(prefers-color-scheme:dark){${styleDataAttributeSelector}{${cssStringFromTheme(
```

Comment:
Any value in useMemo-ing anything here? Haven't looked a lot at how often these things tend to render or how expensive these calls are, mostly just curious

Reply:
Turning all of the theme variables into a string seems to be pretty inexpensive for two themes, I don't think we need to memo this quite yet

<img width="630" alt="image" src="https://github.com/MystenLabs/sui/assets/7453188/2933fb4e-0be0-41b4-9e7f-8100a1406995">